activemq-stomp/src/main/java/org/apache/activemq/transport/stomp/StompWireFormat.java

```diff
@@ -189,29 +188,52 @@ private String readLine(DataInput in, int maxLength, String errorMessage) throws
     }
 
     private ByteSequence readHeaderLine(DataInput in, int maxLength, String errorMessage) throws IOException {
+        boolean isEscaping = false;
         byte b;
-        ByteArrayOutputStream baos = new ByteArrayOutputStream(maxLength);
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream(maxLength);
         while ((b = in.readByte()) != '\n') {
-            if (baos.size() > maxLength) {
-                baos.close();
+            if (outputStream.size() > maxLength) {
+                outputStream.close();
                 throw new ProtocolException(errorMessage, true);
             }
-            baos.write(b);
+            if (isEscaping) {
+                if (!isValidEscapedCharacter(b)) {
+                    throw new ProtocolException("Undefined escape sequence [\\"+((char) (b & 0xFF))+"] found in header!", true);
+                }
+                isEscaping = false;
+            } else if (b == Stomp.ESCAPE) {
+                isEscaping = true;
+            }
+            outputStream.write(b);
```

Comment:
what about trailing "\\"?

Reply:
What's a trailing escape? In this context, these are headers, so there must be a body that follows

Reply:
I was wondering what if line 205 is executed and the next byte is '\n'? Can a legit header value end with an unescaped backslash?

Reply:
Yeah it should be just a backslash then. There's nothing in the spec to prevent that.